### PR TITLE
[JENKINS-43472] Add Maven 3.5.0 installation.

### PR DESCRIPTION
**Pull request by cyrille-leclerc**
[JENKINS-43472](https://issues.jenkins-ci.org/browse/JENKINS-43472) Add Maven 3.5.0 installation.

Successfully tested in https://github.com/jenkinsci/pipeline-maven-plugin/pull/33

ℹ️ instead of adding the binaries to the git repository, we use the maven-dependency-plugin to get maven-3.5.0 from Maven Central. This approach could be used to remove large binary files from the git repo (apache-maven 3.0.1 and 3.1.0)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -81,4 +82,35 @@ THE SOFTWARE.
...
 
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <version>3.0.0</version>
+                <executions>
+                    <execution>
+                        <id>copy-maven-spy-jar</id>
```
Comment (jglick) [inline pom.xml]: What?

**Review by jglick — CHANGES_REQUESTED** [review]

**Comment by jglick** [comment]
> This approach could be used to remove large binary files from the git repo (apache-maven 3.0.1 and 3.1.0)

Why not do that now too?

**Comment by cyrille-leclerc** [comment]
>> This approach could be used to remove large binary files from the git repo (apache-maven 3.0.1 and 3.1.0)

> Why not do that now too?

Done

File: pom.xml (hunk 1 above)
Comment (cyrille-leclerc) [reply pom.xml]: Fixed

**Comment by cyrille-leclerc** [comment]
Note urgent as I have implemented a "workaround" in the pipeline-maven-project:

* https://github.com/jenkinsci/pipeline-maven-plugin/blob/87568088f87b5da4e5fbf36a0ccafd3fc1f785f0/jenkins-plugin/src/test/java/org/jvnet/hudson/test/ExtendedToolInstallations.java
* https://github.com/jenkinsci/pipeline-maven-plugin/blob/87568088f87b5da4e5fbf36a0ccafd3fc1f785f0/jenkins-plugin/pom.xml#L253

**Comment by cyrille-leclerc** [comment]
@reviewbybees

**Review by jglick — APPROVED** [review]

**Comment by ghost** [comment]
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

**Comment by cyrille-leclerc** [comment]
@jglick could we release this plugin?

**Comment by jglick** [comment]
Tried to release 2.1 but the connection timed out because it was so gigantic. Noted in a new README that I think this repository needs to be deleted and the use case reconsidered.

**Pull request merged**